scipy/io/wavfile.py

```diff
@@ -489,54 +519,124 @@ def _handle_pad_byte(fid, size):
+
+    **Left-justified format**
+
+    Data is returned in the smallest compatible numpy int type.  8-bit and
+    lower is unsigned, while 9-bit and higher is signed.
+
+    For example, 24-bit data will be stored as int32, with the MSB of the
+    24-bit data stored at the MSB of the int32, and typically the least
+    significant byte is 0x00.  (However, if a file actually contains data past
+    its specified bit depth, those bits will be read and output, too. [2]_)
+
+    This bit justification and sign matches WAV's native internal format, which
+    allows memory mapping of WAV files that use 1, 2, 4, or 8 bytes per sample
+    (so 24-bit files cannot be memory-mapped, but 32-bit can).
+
+    IEEE float PCM in 32- or 64-bit format is supported, with or without mmap.
+    Values exceeding [-1, +1] are not clipped.
 
     Common data types: [1]_
 
     =====================  ===========  ===========  =============
          WAV format            Min          Max       NumPy dtype
     =====================  ===========  ===========  =============
     32-bit floating-point  -1.0         +1.0         float32
-    32-bit PCM             -2147483648  +2147483647  int32
-    16-bit PCM             -32768       +32767       int16
-    8-bit PCM              0            255          uint8
+    32-bit integer PCM     -2147483648  +2147483647  int32
+    24-bit integer PCM     -2147483648  +2147483392  int32
+    16-bit integer PCM     -32768       +32767       int16
+    8-bit integer PCM      0            255          uint8
     =====================  ===========  ===========  =============
 
-    Note that 8-bit PCM is unsigned.
+    **Right-justified format**
+
+    Data is returned in the smallest compatible numpy int type, and is always
+    signed.
+
+    For example, 24-bit data will be stored as int32, with the LSB of the
+    24-bit data stored at the LSB of the int32, and the MSB as 0x00 or 0xff,
+    depending on sign.  (If a file actually contains data past its specified
+    bit depth, those bits will be discarded during the shift. [2]_)
+
+    Examples:
+
+    =====================  ===========  ===========  =============
+         WAV format            Min          Max       NumPy dtype
+    =====================  ===========  ===========  =============
+    32-bit integer PCM     -2147483648  +2147483647  int32
+    24-bit integer PCM     -8388608     +8388607     int32
+    16-bit integer PCM     -32768       +32767       int16
+    12-bit integer PCM     -2048        +2047        int16
+    8-bit integer PCM      -128         +127         int8
+    =====================  ===========  ===========  =============
+
+    **Fixed-point format**
+    This converts the integer values to float, dividing by 2**(N-1), which
+    interprets them as fixed-point numbers with the binary point
+    to the right of the MSbit, so it can reach -1.0 but cannot reach +1.0.
+    [3]_ Android [4]_ USB?
```

Comment:
[3]_ [4]_ [5]_ [6]_ refs are not in the References section (causing CircleCI failure)

Reply:
Sorry, these are all just notes to people reading the draft, it's not meant to pass tests yet